mount/mount.go

```diff
@@ -22,3 +47,48 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// TempMount mounts the provided mounts to a temp dir, and pass the temp dir to f.
+// The mounts are valid during the call to the f.
+// Finally we will unmount and remove the temp dir regardless of the result of f.
+func TempMount(mounts []Mount, m Mounter, f func(root string) error) error {
+	if len(mounts) == 0 {
+		return errdefs.ErrInvalidArgument
+	}
```

Comment:
I believe this is a change in behaviour. Prevously `len(mounts) == 0` would have been just a no-op.

Is this error necessary? Could this instead just `return nil` ?

Reply:
agree. We give a empty dir to user.

Reply:
done